contracts/astroport-liquid-pooler/src/contract.rs

```diff
@@ -0,0 +1,464 @@
+            funds: vec![a_coin, b_coin],
+        }),
+        DOUBLE_SIDED_REPLY_ID,
+    )))
+}
+
+/// attempts to build a single sided `ProvideLiquidity` message.
+/// pool ratio does not get validated here. as long as the single
+/// side asset amount being provided is within our predefined
+/// single-side liquidity limits, we provide it.
+fn try_get_single_side_lp_submsg(
+    deps: DepsMut,
+    coin_a: Coin,
+    coin_b: Coin,
+    lp_config: LpConfig,
+    asset_data: AssetData,
+) -> Result<Option<SubMsg>, ContractError> {
+    let holder_address = match HOLDER_ADDRESS.may_load(deps.storage)? {
+        Some(addr) => addr,
+        None => return Err(ContractError::MissingHolderError {}),
+    };
+    
+    let assets = asset_data.to_asset_vec(coin_a.amount, coin_b.amount);
+
+    // given one non-zero asset, we build the ProvideLiquidity message
+    let single_sided_liq_msg = ProvideLiquidity {
+        assets,
+        slippage_tolerance: lp_config.slippage_tolerance,
+        auto_stake: lp_config.autostake,
+        receiver: Some(holder_address.to_string()),
+    };
+
+    // now we try to submit the message for either B or A token single side liquidity
+    if coin_a.amount.is_zero()
+        && coin_b.amount <= lp_config.single_side_lp_limits.asset_b_limit
+    {
+        // update the provided liquidity info
+        PROVIDED_LIQUIDITY_INFO.update(deps.storage, |mut info| -> StdResult<_> {
+            info.provided_amount_b = info.provided_amount_b.checked_add(coin_b.amount)?;
+            Ok(info)
+        })?;
+
+        // if available ls token amount is within single side limits we build a single side msg
+        let submsg = SubMsg::reply_on_success(
+            CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: lp_config.pool_address.to_string(),
+                msg: to_binary(&single_sided_liq_msg)?,
+                funds: vec![coin_b],
+            }),
+            SINGLE_SIDED_REPLY_ID,
+        );
+
+        return Ok(Some(submsg));
+    } else if coin_b.amount.is_zero()
+        && coin_a.amount <= lp_config.single_side_lp_limits.asset_a_limit
+    {
+        // update the provided liquidity info
+        PROVIDED_LIQUIDITY_INFO.update(deps.storage, |mut info| -> StdResult<_> {
+            info.provided_amount_a =
+                info.provided_amount_a.checked_add(coin_a.amount)?;
+            Ok(info)
+        })?;
+
+        // if available A token amount is within single side limits we build a single side msg
+        let submsg = SubMsg::reply_on_success(
+            CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: lp_config.pool_address.to_string(),
+                msg: to_binary(&single_sided_liq_msg)?,
+                funds: vec![coin_a],
+            }),
+            SINGLE_SIDED_REPLY_ID,
+        );
+
+        return Ok(Some(submsg));
+    }
+
+    // if neither a nor b token single side lp message was built, we just go back and wait
```

Comment:
What if the amount is higher then `single_side_lp_limit`? in this code the funds will be stuck until we unstuck them using a migration